Tests/tests/DataStore2.spec.lua

```diff
@@ -79,6 +79,44 @@ return function()
 				expect(DataStore2(nonNilKey, fakePlayer):Get("badDefault")).to.equal("abc")
 			end)
 
+			it("should validate set", function()
+				local dataStore = DataStore2(UUID(), fakePlayer)
+				local function testValidator(dataToValidate)
+					if dataToValidate == "yepp" then
+						return true
+					elseif dataToValidate == "definitelyNot" then
+						error("A validation error message")

```

Comment:
This isn't how validators should work, and I'm not sure you ran these tests.

Validators should return true/false and an optional error message, not error themselves.